### PR TITLE
Fix V1 feature flags

### PR DESCRIPTION
**Pull request by jsdw**
And add a small example.

I'll be working on V2 after this, but thought I'd get this in because it at least improves things in V1 land a little.

### CODE REVIEW

File: bin/archive-demo/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+	pretty_env_logger::init();
+
+	// Connect to a node with an RPC client:
+	let rpc_client = RpcClient::from_url("wss://rpc.polkadot.io").await?;
+	let methods = LegacyRpcMethods::<PolkadotConfig>::new(rpc_client.clone());
+
+	let mut bn = 1;
```
Comment (niklasad1) [inline bin/archive-demo/src/main.rs:21]: I think `block_number` is easier to understand here,

My initial take was `big number` but clearly wrong :P

File: bin/tx-decoder/Cargo.toml [hunk 2]
```diff
@@ -12,21 +12,21 @@ rust-version.workspace = true
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-desub = { workspace = true, features = [ "polkadot-js" ] }
+desub = { workspace = true, features = ["polkadot-js"] }
```
Comment (niklasad1) [inline bin/tx-decoder/Cargo.toml:15]: :ok_hand:

File: desub-json-resolver/Cargo.toml [hunk 3]
```diff
@@ -18,9 +18,8 @@ log = { workspace = true }
...
 syn = { workspace = true, features = ["parsing", "derive"] }
-phf = { workspace = true, features = [ "macros" ] }
+phf = { workspace = true, features = ["macros"] }
 
 [features]
-default = ["default_definitions"]
-polkadot = []
-default_definitions = []
+default = ["default-definitions"]
+default-definitions = []
```
Comment (niklasad1) [inline desub-json-resolver/Cargo.toml:25]: :ok_hand:

**Review by niklasad1 — APPROVED** [review]

File: bin/archive-demo/src/main.rs (hunk 1 above)
Comment (jsdw) [reply bin/archive-demo/src/main.rs]: Agreed :)

**Pull request merged**